clang/test/CodeGenSYCL/stall_enable.cpp

```diff
@@ -0,0 +1,32 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+[[intel::stall_enable]] void test() {}
+
+class Foo {
+public:
+  [[intel::stall_enable]] void operator()() const {}
+};
+
+int main() {
+  q.submit([&](handler &h) {
+    Foo f;
+    h.single_task<class test_kernel1>(f);
+
+    h.single_task<class test_kernel2>(
+        []() { test(); });
```

Comment:
I am confused by this case.   Is the attribute on function test(), expected to be propagated?

Reply:
No, it should not be propagated.